### PR TITLE
Site Editor: Initial routing refactoring to separate preview from list view

### PR DESCRIPTION
**Pull request by youknowriad**
Related #55083 

## What?

The List data views (either pages or templates) render two "areas" in the site editor:

 - One area rendering the list of items.
 - One area rendering the preview of the currently selected item.

In trunk, these two areas are rendered by the same component within the "frame". The problem with that approach is that we have other routes where the "preview" is still there but not the "content" area and we want seamless transitions and animations between all of these pages.

This means that we need to have separate the "preview" area from the "content" area and potentially keep the same "preview" area between different routes regardless of whether they use "list" layout or not.

This PR implements this and can also be considered a start of a more formal routing API (later down the road) in the site editor. Note that I didn't touch the sidebar on purpose, the fact that we're still relying on an uncontrolled "Navigator" component makes it hard to refactor as a "routing area".

## Testing Instructions

1- Navigate in the site editor (everywhere :P) with both the "dataviews" experiment enabled and disabled.

### CODE REVIEW

File: packages/edit-site/src/components/layout/index.js [hunk 1]
```diff
@@ -296,82 +281,82 @@ export default function Layout() {
...
-															transition: {
-																duration:
-																	disableMotion
-																		? 0
-																		: 0.5,
-																ease: 'easeOut',
-															},
-													  }
-													: {}
+					{ areas.content && canvasMode !== 'edit' && (
```
Comment (youknowriad) [inline packages/edit-site/src/components/layout/index.js:283]: This is the new "content" area.

File: packages/edit-site/src/components/layout/index.js [hunk 2]
```diff
@@ -296,82 +281,82 @@ export default function Layout() {
 
 					<SavePanel />
 
-					{ showCanvas && (
-						<>
-							{ isListPage && <PageMain /> }
```
Comment (youknowriad) [inline packages/edit-site/src/components/layout/index.js:301]: The list page is now rendered within the "content" area while the editor is rendered within the "preview" area.

File: packages/edit-site/src/components/layout/router.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import Editor from '../editor';
+import DataviewsPatterns from '../page-patterns/dataviews-patterns';
+import PagePages from '../page-pages';
+import PagePatterns from '../page-patterns';
+import PageTemplateParts from '../page-template-parts';
+import PageTemplates from '../page-templates';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+export default function useLayoutAreas() {
```
Comment (youknowriad) [inline packages/edit-site/src/components/layout/router.js:20]: This is what you could describe as "route definitions" that would be easy to swap later with a formal router API. Right now it has no impact on the sidebar but we could make the sidebar routing part of this hook when we make the Navigator component "controllable".

File: packages/edit-site/src/components/layout/router.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	const isSiteEditorLoading = useIsSiteEditorLoading();
+	const { params } = useLocation();
+	const { postType, postId, path, type, isCustom } = params ?? {};
+
+	// Regular page
+	if ( path === '/page' ) {
+		const isListLayout =
+			isCustom !== 'true' && ( ! type || type === 'list' );
```
Comment (youknowriad) [inline packages/edit-site/src/components/layout/router.js]: This is the difficulty here, basically the route definition relies on the "type" of the dataview (layout), which means I had to add this information to the URL. 

There's maybe another way to solve this but in trunk the "router" can't access this information because it's only kept in memory.

File: packages/edit-site/src/components/layout/style.scss [hunk 5]
```diff
@@ -300,3 +293,16 @@
 	}
 
 }
+
+.edit-site-layout__area {
```
Comment (youknowriad) [inline packages/edit-site/src/components/layout/style.scss:291]: This is the style that make the area look like a rounded frame. This style maybe be shared with the "preview" as well.

File: packages/edit-site/src/components/page-pages/index.js [hunk 6]
```diff
@@ -38,21 +37,31 @@ import {
...
+	const selectedDefaultView = useMemo( () => {
+		const defaultView =
+			isCustom === 'false' &&
+			DEFAULT_VIEWS[ postType ].find(
+				( { slug } ) => slug === activeView
+			)?.view;
+		if ( isCustom === 'false' && type ) {
+			return {
+				...defaultView,
+				type,
```
Comment (youknowriad) [inline packages/edit-site/src/components/page-pages/index.js]: We override the default view with the "type" that comes from the URL.

File: packages/edit-site/src/components/page-pages/index.js [hunk 7]
```diff
@@ -107,13 +116,26 @@ function useView( type ) {
...
 
+	const setDefaultViewAndUpdateUrl = useCallback(
+		( viewToSet ) => {
+			if ( viewToSet.type !== view?.type ) {
+				history.push( {
+					...params,
+					type: viewToSet.type,
+				} );
+			}
+			setView( viewToSet );
```
Comment (youknowriad) [inline packages/edit-site/src/components/page-pages/index.js:127]: In addition to updating the view object, we also update the url if the "view type" changes. Maybe a better way is to split out the type from the view state.

File: packages/edit-site/src/components/page-pages/index.js [hunk 8]
```diff
@@ -131,12 +153,20 @@ const DEFAULT_STATUSES = 'draft,future,pending,private,publish'; // All but 'tra
...
+	const { isCustom = 'false' } = params;
 
 	const onSelectionChange = useCallback(
-		( items ) => setPageId( items?.length === 1 ? items[ 0 ].id : null ),
-		[ setPageId ]
+		( items ) => {
+			if ( isCustom === 'false' && view?.type === LAYOUT_LIST ) {
+				history.push( {
+					...params,
+					postId: items.length === 1 ? items[ 0 ].id : undefined,
```
Comment (youknowriad) [inline packages/edit-site/src/components/page-pages/index.js:165]: The selection change persists in the url which allow us to update the "preview" area with the right postId.

File: packages/edit-site/src/components/page-pages/index.js [hunk 9]
```diff
@@ -347,85 +377,33 @@ export default function PagePages() {
...
-					actions={ actions }
-					data={ pages || EMPTY_ARRAY }
-					isLoading={ isLoadingPages || isLoadingAuthors }
-					view={ view }
-					onChangeView={ onChangeView }
-					onSelectionChange={ onSelectionChange }
-					onDetailsChange={ onDetailsChange }
-				/>
-			</Page>
-			{ view.type === LAYOUT_LIST && (
```
Comment (youknowriad) [inline packages/edit-site/src/components/page-pages/index.js:384]: All what's after that has been removed, it now corresponds to the default "preview" area rendered in the Layout.

**Comment by oandregal** [comment]
I'm starting to look into the PR and familiarizing myself with a few things. I've noticed that custom views don't seem to update the `type` param:

https://github.com/WordPress/gutenberg/assets/583546/fd562416-103d-4eac-9a59-ec4313f718a4

**Comment by youknowriad** [comment]
@oandregal Yes, I did that on purpose, basically, the "list" view on custom views is just a full width list view without previews. I think that's fine for a start. We have options to address that, I just wanted to keep it simple for now.

File: packages/edit-site/src/components/layout/index.js (hunk 1 above)
Comment (oandregal) [reply packages/edit-site/src/components/layout/index.js:283]: For someone with limited exposure to site editor code, like myself, the fact that the logic to decide what to render was split in so many places made it hard to navigate and understand. I welcome an approach that centralizes the routing logic and maps it to render areas.

With that in mind, why the `canvasMode` shouldn't be absorbed by the `useLayout` hook?

Comment (youknowriad) [reply packages/edit-site/src/components/layout/index.js:283]: > With that in mind, why the canvasMode shouldn't be absorbed by the useLayout hook?

Can you clarify a bit more what you have in mind?

File: packages/edit-site/src/components/layout/router.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+				content:
+					window.__experimentalAdminViews && isListLayout
+						? 380
+						: undefined,
+			},
+		};
+	}
+
+	// Regular other post types
+	if ( postType && postId ) {
```
Comment (oandregal) [inline packages/edit-site/src/components/layout/router.js:49]: This is the details view, as it exists today. As a follow-up, we'll need to update this to absorb the details as part of the content area (it's now part of the sidebar). ([Mockup](https://github.com/WordPress/gutenberg/pull/57523)).

Comment (youknowriad) [reply packages/edit-site/src/components/layout/router.js:49]: Indeed, we might need to have it specific per post type. Remains to be seen.

File: packages/edit-site/src/components/layout/router.js [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+				content: window?.__experimentalAdminViews ? (
+					<DataviewsPatterns />
+				) : (
+					<PagePatterns />
+				),
+			},
+		};
+	}
+
+	// Fallback shows the home page preview
```
Comment (oandregal) [inline packages/edit-site/src/components/layout/router.js:99]: This is rendered when path is `/wp_template`, `/wp_global_styles`, `/navigation`.

Comment (youknowriad) [reply packages/edit-site/src/components/layout/router.js:99]: yes, and the home page of the site editor too.

File: packages/edit-site/src/components/layout/router.js (hunk 4 above)
Comment (oandregal) [reply packages/edit-site/src/components/layout/router.js]: Why do you have concerns about having the `type` as a URL param?

This makes me consider a more general point: why routing can't act as a pure function of the URL? Given a URL, I'm able to share it with other person who will see the same screen I'm seeing. This, of course, extends to things other than the view type (filters/search, pagination, etc.).

**Comment by oandregal** [comment]
https://github.com/WordPress/gutenberg/pull/57933 landed so I rebased this PR and pushed https://github.com/WordPress/gutenberg/pull/57938/commits/0bdacb3426c3561ab148c5fcd1e59f0028304bb2 for templates.

File: packages/edit-site/src/components/layout/index.js (hunk 1 above)
Comment (oandregal) [reply packages/edit-site/src/components/layout/index.js:283]: It'd be great to see something along these lines:

```js
// The useLayoutAreas hook centralizes all logic about which area is avaliable.
function useLayoutAreas() {
    if ( path === '/page' && canvasMode === 'edit' ) {
        return { areas: { /* ... */ } };
    }

    if ( path === '/page' ) {
        return { areas: { /* ... */ } };
    }

    // etc.
}

// The Layout component composes the layout.
export default function Layout() {
    const { areas } = useLayoutAreas();

    return (
        <div className="edit-site-layout">
            { areas.sidebar && <div className="edit-site-layout__sidebar">{ areas.sidebar }</div> }
            { areas.content && <div className="edit-site-layout__content">{ areas.sidebar }</div> }
            { areas.preview && <div className="edit-site-layout__preview">{ areas.sidebar }</div> }
        </div>
    );
}
```

File: packages/edit-site/src/components/layout/router.js (hunk 4 above)
Comment (youknowriad) [reply packages/edit-site/src/components/layout/router.js]: I think the unknown here is that most of the views could either become entities in the future or be persisted in preferences or things like that, which means the "type" in the url redundant and something we'd want to ditch.

File: packages/edit-site/src/components/layout/index.js (hunk 1 above)
Comment (youknowriad) [reply packages/edit-site/src/components/layout/index.js:283]: I agree that's the goal. 

 - We can't do sidebar now for different reasons: the navigation component need to be controlled (which is not at the moment).
 - We can't ditch the canvas mode entirely (maybe later?) because it's a state that is used in very different places.

**Comment by youknowriad** [comment]
@oandregal maybe you missed a small change to `use-init-edited-entity-from-url.js` for the path check as well (now that templates can be list view by default).

File: packages/edit-site/src/components/layout/router.js (hunk 4 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/layout/router.js]: Are we okay with showing the front page when we have no selection in `pages list`? Personally I think this a bit confusing..

Comment (oandregal) [reply packages/edit-site/src/components/layout/router.js]: > Are we okay with showing the front page when we have no selection in pages list? Personally I think this a bit confusing..

The way I see it, it's just how the (rest of the) site editor works, so it makes sense that pages does the same.

File: packages/edit-site/src/components/page-patterns/style.scss [hunk 12]
```diff
@@ -1,10 +1,11 @@
 .edit-site-patterns {
+	background: #1e1e1e;
 	border-left: 1px solid $gray-800;
-	background: none;
 	margin: $header-height 0 0;
 	border-radius: 0;
 	padding: 0;
 	overflow-x: auto;
+	min-height: 100%;
```
Comment (ntsekouras) [inline packages/edit-site/src/components/page-patterns/style.scss:8]: There is something wrong with the styles in patterns page with `no` experiment enabled and you can't scroll. I haven't found the culprit yet..

File: packages/edit-site/src/components/page-templates/index.js [hunk 13]
```diff
@@ -156,18 +154,30 @@ function TemplatePreview( { content, viewType } ) {
 }
 
 export default function DataviewsTemplates() {
-	const [ templateId, setTemplateId ] = useState( null );
-	const [ view, setView ] = useState( DEFAULT_VIEW );
+	const { params } = useLocation();
+	const { type } = params;
```
Comment (ntsekouras) [inline packages/edit-site/src/components/page-templates/index.js]: I feel the `type` is way too generic and needs rename. Also it would be good to reset it on other routes. I think we are(/were?) resetting some values on history change..

File: packages/edit-site/src/components/layout/router.js (hunk 4 above)
Comment (youknowriad) [reply packages/edit-site/src/components/layout/router.js]: I agree that it's not entirely clear whether the "home page" is the right choice there. A placeholder (select item kind of thing) could work like emails but I agree that for now it's probably fine. I think it would be good to show bulk actions there somehow in case of multi-selections... But all of these can be seen as follow-up improvements.

File: packages/edit-site/src/components/page-patterns/style.scss (hunk 12 above)
Comment (youknowriad) [reply packages/edit-site/src/components/page-patterns/style.scss:8]: Yes, I was wondering whether I should fix it or not, especially because we're stabilizing that page very soon. Just waiting for the "click on preview" to edit.

File: packages/edit-site/src/components/page-templates/index.js (hunk 13 above)
Comment (youknowriad) [reply packages/edit-site/src/components/page-templates/index.js]: Added the reset but yet, "type" is not perfect. Maybe "layout"?

File: packages/edit-site/src/components/layout/router.js [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+		const isListLayout =
+			isCustom !== 'true' && ( ! type || type === 'list' );
+		return {
+			areas: {
+				content: <PageTemplates />,
+				preview: isListLayout && (
+					<Editor isLoading={ isSiteEditorLoading } />
+				),
+			},
+			widths: {
```
Comment (ntsekouras) [inline packages/edit-site/src/components/layout/router.js:71]: The concept of widths in a router 'API' is weird. I'm wondering if we could apply this in the `content` prop and have some css rules in `edit-site-layout__area` container like `width: max-content` or something(I haven't tried anything to see if it's possible with css).

Comment (youknowriad) [reply packages/edit-site/src/components/layout/router.js:71]: It's possible but the reason I kept it here for now is that I thought that maybe the "resizing" that is implemented in layout could use that width at some point.

**Review by ntsekouras — APPROVED** [review]
I think this is a good start and we can iterate. One important thing though is the comment for current patterns page([no scroll](https://github.com/WordPress/gutenberg/pull/57938#discussion_r1457443227)). With a quick follow up tomorrow, we can update the patterns page too I guess.

I'll approve, but since there are too many things to test(different views with and without the experiment), I'd appreciate some more testing from others too.

File: packages/edit-site/src/components/page-templates/index.js (hunk 13 above)
Comment (oandregal) [reply packages/edit-site/src/components/page-templates/index.js]: That's a good point. We could go with `layout` or `viewLayout`.

Comment (oandregal) [reply packages/edit-site/src/components/page-templates/index.js]: Lol. I commented without refreshing the PR and just saw that it was changed ot `layout`. 👍

File: packages/edit-site/src/components/layout/index.js [hunk 15]
```diff
@@ -154,7 +138,7 @@ export default function Layout() {
 	// Sets the right context for the command palette
 	let commandContext = 'site-editor';
 
-	if ( canvasMode === 'edit' && isEditorPage ) {
```
Comment (oandregal) [inline packages/edit-site/src/components/layout/index.js:157]: I looked around and haven't found any page with `canvasMode=edit` that wasn't in the editor.

Comment (oandregal) [reply packages/edit-site/src/components/layout/index.js:157]: Also interacted with the command palette in various places, and it seems to work as expected.

File: packages/edit-site/src/components/layout/index.js [hunk 16]
```diff
@@ -228,7 +212,7 @@ export default function Layout() {
 					/>
 
 					<AnimatePresence initial={ false }>
-						{ isEditorPage && isEditing && (
+						{ canvasMode === 'edit' && (
```
Comment (oandregal) [inline packages/edit-site/src/components/layout/index.js:215]: Same as above: didn't find any page with `canvasMode=edit` that wasn't the editor.

File: packages/edit-site/src/components/layout/index.js [hunk 17]
```diff
@@ -272,7 +256,7 @@ export default function Layout() {
 						className="edit-site-layout__sidebar-region"
 					>
 						<AnimatePresence>
-							{ showSidebar && (
+							{ canvasMode === 'view' && (
```
Comment (oandregal) [inline packages/edit-site/src/components/layout/index.js:259]: This change makes the sidebar take the whole screen in small viewports for pages and templates, while before it showed the content area. Note that showing the sidebar is the same behavior that navigation, styles, and patterns have in `trunk` right now. I think it's fine to go with this. The previous logic based on listing/editor pages was hard to follow and created inconsistencies.

File: packages/edit-site/src/components/layout/index.js [hunk 18]
```diff
@@ -296,82 +280,82 @@ export default function Layout() {
...
+							style={ {
+								maxWidth: widths?.content,
+							} }
+						>
+							{ areas.content }
+						</div>
+					) }
+
+					{ areas.preview && (
+						<div className="edit-site-layout__canvas-container">
```
Comment (oandregal) [inline packages/edit-site/src/components/layout/index.js:295]: I'd appreciate it if we could consolidate the classes around the concept of "areas" as well. These are being used in a few places, so it can be done in a follow-up as well.

**Comment by oandregal** [comment]
_Edit 2_: I can no longer reproduce this issue, either in `trunk` or this branch. I attribute it to a glitch in the matrix.

~_Edit_: I can reproduce the same in `trunk`~ 🤔 

~I'm running into this issue: every time the editor is instantiated, I cannot go back. It happens with the experiment enabled or disabled:~

https://github.com/WordPress/gutenberg/assets/583546/82a743c0-7be9-4d3a-bdec-60bc809f8f24

File: packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js [hunk 19]
```diff
@@ -189,12 +195,25 @@ function useResolveEditedEntityAndContext( { postId, postType } ) {
 			return { postType, postId };
 		}
 
+		// Some URLs in list views are different
```
Comment (oandregal) [inline packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js:198]: I'm a bit confused with the logic of this hook. Specifically as to why sometimes we set the context with `postType`/`postId` and other times we set `postType`/`postId` directly. This is a bit outside the scope of this PR, so I won't block it – though I'd welcome some pointer for me to learn up (and possibly ways to improve in follow-ups).

I've tested that the Pages and Templates with `postId` (list view with item selected) work fine.

File: packages/edit-site/src/components/page-patterns/style.scss (hunk 12 above)
Comment (oandregal) [reply packages/edit-site/src/components/page-patterns/style.scss:8]: The issue with scroll is larger: I cannot scroll _any_ content area (patterns, templates, pages) with or without the experiment enabled. For example, this is how it works in `trunk`:

https://github.com/WordPress/gutenberg/assets/583546/94c71fb1-6ddf-498d-9671-4b147317218a

File: packages/edit-site/src/components/page-pages/index.js [hunk 20]
```diff
@@ -10,7 +10,6 @@ import { dateI18n, getDate, getSettings } from '@wordpress/date';
 import { privateApis as routerPrivateApis } from '@wordpress/router';
 import { useSelect, useDispatch } from '@wordpress/data';
 import { DataViews } from '@wordpress/dataviews';
-import { ENTER, SPACE } from '@wordpress/keycodes';
```
Comment (oandregal) [inline packages/edit-site/src/components/page-pages/index.js:13]: I can no longer access the editor via the keyboard in any page. Once the preview is focused, the ENTER key doesn't trigger the edit canvas mode. SPACE triggers a scroll.

https://github.com/WordPress/gutenberg/assets/583546/b49cd183-0fcc-4633-b790-a1deb03b1508

Interestingly, in `trunk`, this behavior was only working in dataviews-powered pages, as far as I've tested.

Comment (oandregal) [reply packages/edit-site/src/components/page-pages/index.js:13]: I remember having to add an extra div to the `PostPreview` editor wrapper to make this work. Perhaps it was never considered before dataviews? If so, it could be a follow-up to add this behavior to the preview area in any page.

File: packages/edit-site/src/components/page-pages/index.js (hunk 8 above)
Comment (oandregal) [reply packages/edit-site/src/components/page-pages/index.js:165]: This is nice. As a follow-up, we should consider updating the content area as well (update the selection).

**Comment by oandregal** [comment]
> One important thing though is the comment for current patterns page(https://github.com/WordPress/gutenberg/pull/57938#discussion_r1457443227). With a quick follow up tomorrow, we can update the patterns page too I guess.

@youknowriad @ntsekouras the scrolling issue is larger: it happens for any content area (cannot scroll any dataviews layouts, for example), with and without the experiment enabled. Given the time constraints, I'm fine if we want to merge this PR for us to start several follow-ups in parallel, but the issue is a bit severe. Other than that, I haven't found any blocker.

The direction and changes are great 👏

File: packages/edit-site/src/components/layout/index.js (hunk 16 above)
Comment (youknowriad) [reply packages/edit-site/src/components/layout/index.js:215]: yes, that's the point here. The header is only shown when the canvasMode is edit, I guess your remark is a bit unclear to me.

File: packages/edit-site/src/components/layout/index.js (hunk 17 above)
Comment (youknowriad) [reply packages/edit-site/src/components/layout/index.js:259]: I agree with this remark, I think the previous logic was too complex but I do believe it's better to show the list of "pages" and "templates" directly (not the sidebar), I believe we should try to follow-up with this. Maybe the router needs to be aware of viewport and change the areas accordingly.

File: packages/edit-site/src/components/page-pages/index.js (hunk 8 above)
Comment (youknowriad) [reply packages/edit-site/src/components/page-pages/index.js:165]: you mean making the selection in dataviews controlled right?

File: packages/edit-site/src/components/page-patterns/style.scss (hunk 12 above)
Comment (youknowriad) [reply packages/edit-site/src/components/page-patterns/style.scss:8]: Ok, I'll check this.

File: packages/edit-site/src/components/layout/index.js (hunk 18 above)
Comment (youknowriad) [reply packages/edit-site/src/components/layout/index.js:295]: I agree, the problem is that the "preview" area is a bit complex with animations and stuff so I was afraid of touching it. Could be a follow-up too.

File: packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js (hunk 19 above)
Comment (youknowriad) [reply packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js:198]: So the return of this hook gives the postType and postId that is going to be rendered in the "preview" and the "editor". 

For "pages", that postType and postId is the "template" of the page, not the page directly because it's the root element. So in order to fill the template with the right postId, we provide it as "context" (BlockContext for the editor which ensures blocks like postContent block... will use the right postId).

This kind of relates with the unification efforts between post and site editors. I think ultimately, there are ways to simplify this further but there's some editSite state that forces us to keep this hook for now.

File: packages/edit-site/src/components/page-pages/index.js (hunk 20 above)
Comment (youknowriad) [reply packages/edit-site/src/components/page-pages/index.js:13]: The PostPreview has been replaced with the regular "Editor"/"Preview" that we use in the site editor for all the site editor pages basically. So there's a chance that the bug you're mentioning has always existed for the "frame" of the site editor.

**Comment by youknowriad** [comment]
@oandregal I agree we should fix the scrolling issue before merging.

File: packages/edit-site/src/components/layout/index.js (hunk 16 above)
Comment (oandregal) [reply packages/edit-site/src/components/layout/index.js:215]: This wasn't meant as something you have to address but as something I've checked, and I see as correct in the PR. Sorry for the confusion :)

Comment (youknowriad) [reply packages/edit-site/src/components/layout/index.js:215]: I'm so dumb :P

File: packages/edit-site/src/components/layout/index.js (hunk 17 above)
Comment (oandregal) [reply packages/edit-site/src/components/layout/index.js:259]: Yeah, agree. We should show the content area instead.

File: packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js (hunk 19 above)
Comment (oandregal) [reply packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js:198]: That's helpful, thanks for the context.

**Comment by youknowriad** [comment]
Ok I've fixed the scrolling behavior.

**Comment by t-hamano** [comment]
This PR appears to have introduced some layout regressions. I have summarized what I discovered in #58044.
